### ISSUE TITLE
Fix unexpected error `Unknown value for flag --fix`

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#8301]
### What minimal example or steps are needed to reproduce the bug?

```sh
npm exec stylelint -- --fix "foo.css" 
```

### What minimal configuration is needed to reproduce the bug?

_any configuration_

### How did you run Stylelint?

```sh
npm exec stylelint -- --fix "foo.css" 
```

### Which Stylelint-related dependencies are you using?

`stylelint@16.13.0`

### What did you expect to happen?

I expected `--fix` to work as a boolean flag, as it did before.

### What actually happened?

```
Error: Unknown value for flag `--fix`: `foo.css`. Value must be one of: [`true`, `lax`, `strict`]
```

### Do you have a proposal to fix the bug?

related to: 
- https://github.com/stylelint/stylelint/issues/7497
- https://github.com/sindresorhus/meow/issues/44

Even if `--fix` was the last flag before the list of inputs it worked fine up to version `16.12.0` because `--fix` was a boolean flag.

Now that `--fix` also supports string keywords as inputs the flag has become ambiguous.
It treats the input option as an incorrect flag keyword.

I suspect that I am not alone in having scripts all over with `--fix` right before the inputs 😅 .
Is there anything we can do to improve the current implementation?

### Workaround

Move the `--fix` flag before any other flag or after the list of inputs.

```diff
- npm exec stylelint -- --max-warnings=0 --fix "foo.css" 
+ npm exec stylelint -- --fix --max-warnings=0 "foo.css"
```

```diff
- npm exec stylelint -- --fix "foo.css" 
+ npm exec stylelint -- "foo.css" --fix 
```

**Comment by TobyPerplex** [comment]
> have the same problem.
> 
> we used stylelint with lint-staged. scripts called at pre-commit hook.

Same

**Comment by romainmenke** [comment]
Should we revert the change to the `--fix` flag and configuration option?

Having read through https://github.com/stylelint/stylelint/issues/7497 it is clear that we didn't consider that additions to `--fix` would be a breaking change. I don't think we would have designed that change the same way if we were aware of that.

Given the seemingly large impact of this issue (currently 13 upvotes) I suggest we revert the additions and revisit how users can opt-in to stricter parsing while fixing later.

This removes the stressy part and gives us more time to pick the best option :)

**Comment by ybiquitous** [comment]
Sorry for the late response. For a quick fix, it seems that we just could stop using `choices`:

https://github.com/stylelint/stylelint/blob/8a173d4ad4164aae9160b69b248de573bd2235d9/lib/cli.mjs#L245

**Comment by ybiquitous** [comment]
I've opened PR #8313. If it's approved, we could release the patch version soon.
